### ISSUE TITLE
weight problem

### ISSUE THREAD

**Issue by litongxin666** [dangweili/pedestrian-attribute-recognition-pytorch#16]
weights = torch.zeros(targets_var.shape)
        for i in range(targets_var.shape[0]):
            for j in range(targets_var.shape[1]):
                if targets_var.data.cpu()[i, j] == -1:
                    weights[i, j] = weight_neg[j]
                elif targets_var.data.cpu()[i, j] == 1:

if targets_var.data.cpu()[i, j] == -1: should be changed to targets_var.data.cpu()[i, j] == 0

**Comment by dangweili** [comment]
@litongxin666 I re-config the weight in train_deepmar_resnet50.py, line 344-352

**Comment by abnercloud** [comment]
Hello, @dangweili ,Thanks to the great codebase! I has some confusions as follow:
 
In train_deepmar_resnet50.py, line 355:
    targets_var[targets_var == -1] = 0
What is the meaning of this line?

In evaluate.py, line 45-46:
    pt_result[pt_result>=0] = 1
    pt_result[pt_result<0] = 0 
And, why do we use 0 here? 

In demo.py, line 129, 137:
    if score[0, idx] >= 0:
Why do we think that greater than 0 means this attribute?

**Comment by abnercloud** [comment]
criterion = F.binary_cross_entropy_with_logits
After logits, do sigmoid operation. Therefore, 0 is the boundary?

**Comment by abnercloud** [comment]
What I am confused about is why the label is set to -1 when the data set is loaded.

**Comment by abnercloud** [comment]
> What I am confused about is why the label is set to -1 when the data set is loaded.

Of course, 0 and -1 do not matter.
